src/scripts/generate_vaccination_status_datamart.py

```diff
@@ -0,0 +1,224 @@
+import pandas as pd
+import pathlib
+import datetime
+from utils import read_blob, record_combination_func, write_blob
+from tqdm import tqdm
+
+
+def determine_covid_vax_status(patient_immunizations: pd.DataFrame) -> pd.Series:
+    """
+    Given a patient's COVID immunizations records return a pandas series containing the
+    patient's demographics and the dates when they:
+        1.  Completed an initial COVID vaccination sequence.
+        2.  Received a booster.
+    When either of these dates cannot be determined return null.
+    """
+    vax_codes = {
+        "jj_adult_initial": [212],
+        "moderna_adult": [207],
+        "pfizer_adult": [208, 217],
+        "pfizer_pediatric": [218],
+        "adult_boosters": [207, 221, 208, 217],
+        "pfizer_12to17_boosters": [208, 217],
+        "moderna_range": range(28, 56 + 1),
+        "pfizer_range": range(21, 56 + 1),
+        "jj_booster_delay": 56,
+        "moderna_booster_delay": 150,
+        "pfizer_booster_delay": 150,
+    }
+
+    patient_immunizations.sort_values(by=["occurrenceDateTime"], ascending=True)
+    vaccinated_date = pd.NaT
+    boosted_date = pd.NaT
+    initial_vax_type = ""
+    for idx, vaccination in patient_immunizations.iterrows():
+
+        current_dose_age = (
+            vaccination.occurrenceDateTime - vaccination.birthDate
+        ).days / 365.25
+
+        if pd.isnull(vaccinated_date):
+            # J&J ages 18+
+            if (
+                vaccination.vaccineCode in vax_codes["jj_adult_initial"]
+                and current_dose_age >= 18
+            ):
+                vaccinated_date = vaccination.occurrenceDateTime + datetime.timedelta(
+                    days=14
+                )
+                initial_vax_type = "jj"
+                next_dose_idx = idx + 1
+
+            # Moderna ages 18+
+            if (
+                vaccination.vaccineCode in vax_codes["moderna_adult"]
+                and current_dose_age >= 18
+            ):
+                vaccinated_date = check_mrna_status(
+                    idx,
+                    patient_immunizations,
+                    vax_codes["moderna_adult"],
+                    vax_codes["moderna_range"],
+                )
+                initial_vax_type = "moderna"
+                next_dose_idx = idx + 2
+
+            # Pfizer ages 12 +
+            if (
```

Comment:
Would this be slightly more efficient (and still correct) using `elif`?

Reply:
Yes definitely!

Reply:
Done.